src/backend/utils/adt/jsonpath_exec.c

```diff
@@ -872,10 +903,10 @@ executeItemOptUnwrapTarget(JsonPathExecContext *cxt, JsonPathItem *jsp,
 						found->list = list_make2(found->singleton, NULL);
 						found->singleton = NULL;
 					}
-					else if (!found->list)
-						found->list = list_make1(NULL); /* Since JsonValueList uses a NULL singleton as shortcut, need to manually insert null value into list */
-					else
+					else if (found->list)
 						found->list = lappend(found->list, NULL);
+					else
+						found->list = list_make1(NULL); /* Since JsonValueList uses a NULL singleton as shortcut, need to manually insert null value into list */
```

Comment:
It's not PG format comment
```
					else
                         /* Since JsonValueList uses a NULL singleton as shortcut, need to manually insert null value into list */
						found->list = list_make1(NULL);
```